### PR TITLE
chore: update .gitignore

### PR DESCRIPTION
**Pull request by danstarns**
Adds some exclusion to the gitignore, all these are added when running the test suite. 

Untitled

### CODE REVIEW

**Comment by millsp** [comment]
Curious to see what changed in your `introspection.test.ts.snap`, looks like it could be a snapshot serializer issue on Windows. Would you mind sharing the diff?

For `dev.db`, maybe we can restrict it to the the `src/__tests__` folder only? How did `dev.db` endup in `sdk` and `client`?

**Review by millsp — COMMENTED** [review]
Curious to see what changed in your `introspection.test.ts.snap`, looks like it could be a snapshot serializer issue on Windows. Would you mind sharing the diff?

For `dev.db`, maybe we can restrict it to the the `src/__tests__` folder only? How did `dev.db` endup in `sdk` and `client`?

File: .gitignore [hunk 1]
```diff
@@ -38,3 +38,7 @@ esm
 reproductions/*
 !reproductions/basic-sqlite
 !reproductions/pnpm-workspace.yaml
+
+dev.db
+junit.xml
+introspection.test.ts.snap
```
Comment (janpio) [inline .gitignore:44]: That one should not be necessary.

Comment (Jolg42) [inline .gitignore:44]:
```suggestion
```
Indeed, this one should be kept, what's the diff for you here for this file?

**Comment by danstarns** [comment]
I think this has to do with something on windows but I'm not sure exactly what the issue is, possibly line endings. Git thinks there is diff but visibly the changes are not there.

**Review by aqrln — APPROVED** [review]

**Pull request merged**